src/facade/dragonfly_connection.cc

```diff
@@ -769,6 +776,20 @@ auto Connection::IoLoop(util::FiberSocketBase* peer, SinkReplyBuilder* orig_buil
   do {
     FetchBuilderStats(stats_, orig_builder);
 
+    if (migration_request_) {
+      ProactorBase* dest = migration_request_;
+
+      dispatch_fb_.JoinIfNeeded();
+      DCHECK(dispatch_q_.empty());
+      migration_request_ = nullptr;
+      this->Migrate(dest);
```

Comment:
Please revisit the checks inside `Connection::Migrate`. For example migrating a connection that has pubsub or monitoring enabled is UB because we store its subscription handle in thread-local caches. It has to be migrated or we skip this case all together